pkg/operator/ceph/disruption/clusterdisruption/add.go

```diff
@@ -92,7 +91,7 @@ func Add(mgr manager.Manager, context *controllerconfig.Context) error {
 			if !ok {
 				return false
 			}
-			return pdb.Name == osdPDBAppName && pdb.Status.DisruptionsAllowed == 0
+			return pdb.Name == osdPDBAppName && (pdb.Status.ExpectedPods-pdb.Status.CurrentHealthy) == 1
```

Comment:
Could you add a comment for the purpose of this check? I'm not clear why we want to filter the event in that case.
```
(pdb.Status.ExpectedPods-pdb.Status.CurrentHealthy) == 1
```

Reply:
The goal here is to start the controller when the `main` PDB is updated. 

- Initial check was `pdb.Status.DisruptionsAllowed == 0`. This will start the controller when first OSD pod goes down. 
- With the new change, we initially increment the `maxUnavailable` count if the OSD is down but PGs are healthy. So maxUnavailable will be incremented multiple times during a drain (because ceph might take some time to update the PGs as degraded), so the reconciler gets triggered multiple times.  So `(pdb.Status.ExpectedPods-pdb.Status.CurrentHealthy) == 1` is the same as `pdb.Status.DisruptionsAllowed == 0`, that is, start the controller when first OSD goes down. 

I'll check if there is a better solution for this.

Reply:
Makes sense, could you also add a code comment to explain this check?

Reply:
This is not required anymore. Back to using ` pdb.Status.DisruptionsAllowed == 0`.  So controller will start when first OSD does down.